### ISSUE TITLE
Support Doxygen arbitrary alias\tags settings (example: use '\par' instead of '\param')

### ISSUE THREAD

**Issue by michelleangela** [microsoft/vscode-cpptools#5700]
We have configured Doxygen to use `\par` instead of `\param`, can that somehow also be supported (probably through some setting?).
Also, for some reason we put our Doxygen comments in between the function signature and the opening `{`. Doxygen parses that just fine ... will that work here as well?

_Originally posted by @Boddlnagg in https://github.com/microsoft/vscode-cpptools/issues/658#issuecomment-649722497_

**Comment by michelleangela** [comment]
> Also, for some reason we put our Doxygen comments in between the function signature and the opening `{`. Doxygen parses that just fine ... will that work here as well?

@Boddlnagg 
I'm not sure if I understand what you mean by "in between the function signature and the opening `{`". Could you provide an example for this?

**Comment by Boddlnagg** [comment]
> I'm not sure if I understand what you mean by "in between the function signature and the opening {". Could you provide an example for this?

Sure:
```c++
int foobar(int a)
/// \brief Foos the bar
/// \par a Something
/// \ret Something else (or not)
{
    return a;
}
```

(forgot to mention that we also have defined `ret` as an abbreviation for `return`)

**Comment by Boddlnagg** [comment]
To elaborate on the original request: We also have defined aliases `\in`, `\out`, and `\inout` for `\param[in]`, `\param[out]` and `\param[in,out]`, respectively. So I think the feature would need to support arbitrary alias settings. Or the VSCode extension needs to read and the original Doxygen config and evaluate the `ALIASES` definitions (https://www.doxygen.nl/manual/config.html#cfg_aliases), but I don't know if that is feasible.

**Comment by michelleangela** [comment]
@Boddlnagg 

Thanks for the additional information. 

Please note that our team prioritizes actions on product features with broad customer impact based on telemetry data and/or upvotes of feature requests.

**Comment by bobbrow** [comment]
It looks like there are two requests in here (additional tags, and comment position).  Should we split it into two issues?
Also, does VS Code support [collapsible sections](https://gist.github.com/pierrejoubert73/902cc94d79424356a8d20be2b382e1ab) in the tooltips?  If so, we could potentially place the raw comment in there as a stop gap.

**Comment by J-Fields** [comment]
The formatted doxygen popup only recognizes `\exception` (not `\throw` or `\throws`), and `\throws` isn't even highlighted in the actual text.

**Comment by michelleangela** [comment]
> The formatted doxygen popup only recognizes `\exception` (not `\throw` or `\throws`), and `\throws` isn't even highlighted in the actual text.

To display other Doxygen comment tags, set the setting `C_Cpp.simplifyStructuredComments` to false. This setting was added to resolve issue https://github.com/microsoft/vscode-cpptools/issues/5706.

**Comment by bobbrow** [comment]
This was addressed in 0.29.0
